api/v0/virtual_machines/virtual_machines.proto

```diff
@@ -0,0 +1,178 @@
+ *                                                                            *
+\* -------------------------------------------------------------------------- */
+
+syntax = "proto3";
+
+package aurae.vms.v0;
+
+option go_package = "github.com/aurae-runtime/ae/client/pkg/api/v0/virtual_machines;vmsv0";
+
+service VmService {
+  /// Reserve requested system resources for a new VM.
+  rpc Allocate(VmServiceAllocateRequest) returns (VmServiceAllocateResponse) {}
+
+  /// Free up previously requested resources for an existing VM
+  rpc Free(VmServiceFreeRequest) returns (VmServiceFreeResponse) {}
+
+  /// Start a new VM.
+  rpc Start(VmServiceStartRequest) returns (VmServiceStartResponse) {}
+
+  /// Stop one or more VMs.
+  rpc Stop(VmServiceStopRequest) returns (VmServiceStopResponse) {}
+}
+
+message VmServiceAllocateRequest{
+  VirtualMachine vm = 1;
+}
+
+message VmServiceAllocateResponse{
+  string vm_id = 1;
+}
+
+message VmServiceFreeRequest{}
+message VmServiceFreeResponse{
+  string vm_id = 1;
+}
+
+message VmServiceStartRequest{
+  string vm_id = 1;
+  string executable = 2;
+}
+message VmServiceStartResponse{}
+
+message VmServiceStopRequest{
+  string vm_id = 1;
+}
+message VmServiceStopResponse{}
+
+
+/// An Aurae virtual machine
+message VirtualMachine {
+  string id = 1;
+  MachineConfiguration machine_config = 2;
+  string kernel_img_path = 3;
+  string kernel_args = 4;
+  RootDrive root_drive = 5;
+  repeated DriveMount drive_mounts = 6;
+  repeated NetworkInterface network_interfaces = 7;
+  uint32 timeout_seconds = 8;
+}
+
+/// Message to set the machine config for a  VM
+message MachineConfiguration {
+  /// Specifies the memory size of VM
+  uint32 mem_size_mb = 1;
+
+  /// Specifies the number of vCPUs for the VM
+  uint32 vcpu_count = 2;
+}
+
+/// Message to specify network config for a VM
+message NetworkInterface {
+  /// Specifies the mac address for the the device
+  string mac_address = 1;
+
+  /// Specifies the name of the tap device on the host
+  string host_dev_name = 2;
+
+  /// IPConfig optionally provides static IP configuration that will be configured
+  /// on the VM's internal networking interface. If not specified, no IP
+  /// configuration will be applied to the VM's internal nic automatically.
+  IPConfiguration ipconfig = 3;
+}
+
+/// Message to specify static IP configuration that will be
+/// applied to a  VM's network interface internally
+message IPConfiguration {
+  /// PrimaryAddr specifies, in CIDR notation, the primary address
+  /// and subnet that a network interface will be assigned inside
+  /// the VM.
+  string primary_addr = 1;
+
+  /// GatewayAddr specifies the default gateway that a network interface
+  /// should use inside the VM.
+  string gateway_addr = 2;
+
+  /// Nameservers is a list of nameservers that the VM will be configured
+  /// to use internally. Currently only up to 2 nameservers can be specified
+  /// (any more in the list will be ignored) and configuration is provided
```

Comment:
Will want to remove much of the opinionated pieces on the implementation, until we decide what our defaults are.